roles/kubernetes-apps/metrics_server/templates/metrics-server-deployment.yaml.j2

```diff
@@ -84,12 +84,15 @@ spec:
       volumes:
         - name: tmp
           emptyDir: {}
-{% if not masters_are_not_tainted %}
       tolerations:
+{% if not masters_are_not_tainted %}
```

Comment:
If `masters_are_not_tainted` is true and `metrics_server_extra_tolerations` is not defined, the above `tolerations` becomes empty and that leads an error due to invalid YAML format.

Reply:
I have commit a fix:
add `tolerations` key , if `masters_are_not_tainted` is false or `metrics_server_extra_tolerations` is defined 

With this condition is that ok ?

Reply:
hi @oomichi ,

now, It seems that the CI failed on error outside of this PR

Due to calico  mismatch checksum


`fatal: [instance-1 -> instance-1]: FAILED! => {"attempts": 4, "changed": true, "checksum_dest": null, "checksum_src": "122ff30c76923425b04a5c8915b6a164b813b0fb", "dest": "/builds/kargo-ci/kubernetes-sigs-kubespray/downloads/calico-v3.25.1-kdd-crds/v3.25.1.tar.gz", "elapsed": 1, "msg": "The checksum for /builds/kargo-ci/kubernetes-sigs-kubespray/downloads/calico-v3.25.1-kdd-crds/v3.25.1.tar.gz did not match 361b0e0e6d64156f0e1b2fbfd18d13217d188eee614eec5de6b05ac0deaab372; it was 4d6b6653499f24f80a85a0a7dac28d9571cabfa25356b08f3b438fd97e322e2d.", "src": "/home/kubespray/.ansible/tmp/ansible-tmp-1681554791.7329743-3253-129198476346473/tmpi_gxw4px", "url": "https://github.com/projectcalico/calico/archive/v3.25.1.tar.gz"}
`

https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/4119243029#L3913

Reply:
I have fix calico checksum in this PR #9990

Reply:
@pli01 Thanks for fixing the calico checksum issue on another pull request!